src/vcpkg/tools.cpp

```diff
@@ -62,82 +63,114 @@ namespace vcpkg
         return std::array<int, 3>{*d1.get(), *d2.get(), *d3.get()};
     }
 
-    static Optional<ToolData> parse_tool_data_from_xml(StringView XML, StringView XML_PATH, StringView tool)
+    ExpectedL<std::vector<ToolDataEntry>> parse_tool_data(StringView contents, StringView origin)
     {
+        auto as_json = Json::parse(contents, origin);
+        if (!as_json)
+        {
+            return as_json.error();
+        }
+        auto as_value = std::move(as_json).value(VCPKG_LINE_INFO).value;
+
+        Json::Reader r(origin);
+        auto maybe_tool_data = r.visit(as_value, ToolDataFileDeserializer::instance);
+        if (!r.errors().empty() || !r.warnings().empty())
+        {
+            return r.join();
+        }
+
+        if (auto tool_data = maybe_tool_data.get())
+        {
+            return *tool_data;
+        }
+
+        return msg::format(msgErrorWhileParsingToolData, msg::path = origin);
```

Comment:
```suggestion
        return maybe_tool_data.value_or_exit(VCPKG_LINE_INFO);
```

Any errors in parsing should have been reported by `Json::Reader`, so if we get here vcpkg has a bug and we shouldn't have separate localized messages for bugs.